tools/flang1/flang1exe/semutil2.c

```diff
@@ -3217,29 +3237,43 @@ _constructf90(int base_id, int in_indexast, bool in_array, ACL *aclp)
          */
         if (acs.level && sem.arrfn.sptr > NOSYM &&
             sem.arrfn.return_value == SST_ASTG(stkp)) {
+          int ret_sptr = A_SPTRG(sem.arrfn.return_value);
           acs.func_in_do = TRUE;
-          if (sem.arrfn.call_std) {
-            /* The function call stmt generated in func_call2() was added
-             * outside the loop generated from implied-do loop, we should move
-             * the call stmt into loop.
-             */
-            ast = STD_AST(sem.arrfn.call_std);
-            ast = ast_rewrite_indices(ast);
-            (void)add_stmt(ast);
-            remove_stmt(sem.arrfn.call_std);
-            sem.arrfn.call_std = 0;
+          /* STDs generated in func_call2(semfunc.c) have been move in loop
+           * when processing AC_IDO. We do not need move in sem.arrfn.alloc_std
+           * and sem.arrfn.call_std again.
+           */
```

Comment:
Is it possible to add a check here to assert that the call STDs are indeed in the loop?

Reply:
I think it is difficult to add an assert here.

Reply:
OK.